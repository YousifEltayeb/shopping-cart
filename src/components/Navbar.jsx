import { Link } from "react-router";
import styled from "styled-components";
import PropTypes from "prop-types";
import { ShoppingCart } from "lucide-react";
const Nav = styled.nav`
  padding: 12px;
  display: flex;
  justify-content: space-around;
  background-color: #171616;
`;

const Span = styled.span`
  color: red;
  margin-left: 8px;
`;

const StyleCheckout = styled.div`
  display: flex;
  place-items: center;
`;

const StyleLinks = styled.div`
  display: flex;
  gap: 12px;
  place-items: center;
  align-items: center;
`;
const Img = styled.img`
  width: 150px;
`;
const Navbar = ({ products = [] }) => {
  const numberOfProductsInCart =
    products.filter((product) => product.inCart).length || 0;
  return (
    <Nav data-testid="navbar">
      <Link to="/">
        <Img src="../public/logo.png" alt="Logo" />
      </Link>
      <StyleLinks>
        <Link to="/">Home</Link>
        <Link to="/Store">Store</Link>
        <StyleCheckout>
          <Link to="/Cart" aria-label="Cart">
            Cart
          </Link>
          <Span>
            {" "}
            <ShoppingCart width="24px" />
            <span data-testid="nav-num-of-products">
              {numberOfProductsInCart}{" "}
            </span>
          </Span>
        </StyleCheckout>
      </StyleLinks>
    </Nav>
  );
};
Navbar.propTypes = {
  products: PropTypes.array,
};

export default Navbar;
