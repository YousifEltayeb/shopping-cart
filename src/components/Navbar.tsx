import { Link } from "react-router";
import styled from "styled-components";
import { ShoppingCart } from "lucide-react";
import logo from "../../public/logo.png";
const Nav = styled.nav`
  padding: 12px;
  display: flex;
  justify-content: space-around;
  background-color: #171616;
`;

const Span = styled.span`
  color: red;
  margin-left: 8px;
`;

const StyleCheckout = styled.div`
  display: flex;
  place-items: center;
`;

const StyleLinks = styled.div`
  display: flex;
  gap: 12px;
  place-items: center;
  align-items: center;
`;
const Img = styled.img`
  width: 150px;
`;
const Navbar = ({ context = [{ inCart: false }] }) => {
  const numberOfProductsInCart =
    context.filter((context) => context.inCart).length || 0;
  return (
    <Nav data-testid="navbar">
      <Link to="/">
        <Img src={logo} alt="Logo" />
      </Link>
      <StyleLinks>
        <Link to="/">Home</Link>
        <Link to="/Store">Store</Link>
        <StyleCheckout>
          <Link to="/Cart" aria-label="Cart">
            Cart
          </Link>
          <Span>
            {" "}
            <ShoppingCart width="24px" />
            <span data-testid="nav-num-of-products">
              {numberOfProductsInCart}{" "}
            </span>
          </Span>
        </StyleCheckout>
      </StyleLinks>
    </Nav>
  );
};

export default Navbar;
