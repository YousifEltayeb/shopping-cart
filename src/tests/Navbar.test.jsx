import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import Navbar from "../components/Navbar";
import { MemoryRouter } from "react-router";
const emptyProducts = [];
const nonEmptyProducts = [{ title: "New product added to cart", inCart: true }];
describe("Navbar", () => {
  it("Store and cart links are rendered", () => {
    const { container } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it("Displaying number of products in cart", async () => {
    const { rerender } = render(
      <MemoryRouter>
        <Navbar products={emptyProducts} />
      </MemoryRouter>,
    );
    // first render should be 0
    expect(screen.getByText("0")).toBeVisible();
    rerender(
      <MemoryRouter>
        <Navbar products={nonEmptyProducts} />
      </MemoryRouter>,
    );
    // pushed a product now should be 1
    expect(screen.getByText("1")).toBeVisible();
  });
});
