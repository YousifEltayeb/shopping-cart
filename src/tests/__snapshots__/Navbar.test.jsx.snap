// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > Store and cart links are rendered 1`] = `
<div>
  <nav
    class="sc-FEMpB jpiCxv"
  >
    <a
      data-discover="true"
      href="/"
    >
      <img
        alt="Logo"
        class="sc-dycHyt jTHsTx"
        src="../public/logo.png"
      />
    </a>
    <div
      class="sc-fLDLck eeNEvz"
    >
      <a
        data-discover="true"
        href="/"
      >
        Home
      </a>
      <a
        data-discover="true"
        href="/Store"
      >
        Store
      </a>
      <div
        class="sc-dVBluf jscDOy"
      >
        <a
          aria-label="Cart"
          data-discover="true"
          href="/Cart"
        >
          Checkout
        </a>
        <span
          class="sc-gjcoXW hffLmj"
        >
           
          <svg
            class="lucide lucide-shopping-cart"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="8"
              cy="21"
              r="1"
            />
            <circle
              cx="19"
              cy="21"
              r="1"
            />
            <path
              d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
            />
          </svg>
          0
           
        </span>
      </div>
    </div>
  </nav>
</div>
`;
